### ISSUE TITLE
Ecotone\EventSourcing\LazyProophEventStore::getWrappedConnection(): Return value must be of type Doctrine\DBAL\Driver\PDOConnection|PDO, Illuminate\Database\PDO\Connection returned

### ISSUE THREAD

**Issue by aidan-casey** [ecotoneframework/ecotone#18]
**Ecotone version(s) affected**: 1.30.1

**Description**  
If the example for using an existing connection in the [documentation](https://docs.ecotone.tech/modules/dbal-support#using-existing-connection) ever worked, it is now broken. This is because Illuminate's custom drivers are not accounted for in the `getWrappedConnection()` method (which, by the way, is deprecated).

**How to reproduce**  
```php
$this->app->singleton(DbalConnectionFactory::class, function () {
    return DbalConnection::create(
        DB::connection('connection_name')->getDoctrineConnection()
    );
});
```

I'm not sure if this is something that can necessarily get fixed without changing the approach.

While I'm here, I will say I really like the general approach of the package, but I'm not a huge fan of how much the event sourcing package is tied to Prooph. Would be awesome to see an approach implemented that allows an Eloquent driver to be implemented, etc.

**Comment by dgafka** [comment]
Hey @aidan-casey 

There is integration with Eloquent, which is enabled out of the box, when using laravel package. 
I just realized there was no mention about that in the documentation, so I've [updated that.](https://docs.ecotone.tech/modules/laravel-ddd-cqrs-event-sourcing#eloquent)

However Eloquent is ORM implementation, which does not suite Event Sourcing well, so the integration works for State-Stored Aggregates (Standard ones).
What would be needed instead is a package, that provides abstraction layer over storing and fetching event streams and possible implementation of projections (to track positions). 

I was planning to provide integration with [eventstoredb](https://www.eventstore.com/eventstoredb), so the event sourced package would be abstracted and `pdo-event-store` and `event-store-db` would actually implement that. 

So I am keen to know how does it looks like in Laravel world. If you can point me to the package that would provide abstraction over event streams, I will be happy to to check it out.

**Comment by dgafka** [comment]
I will check the issue and let you know :)

**Comment by aidan-casey** [comment]
@dgafka - I have seen that integration, but my resistance to it is that it ties our domain logic directly to infrastructure code.

It would maybe be helpful to have some documentation on what is needed to provide an adapter that can handle storage of events, snapshots, sagas, etc. This could easily allow someone to create a bridge for Laravel.

The package [spatie/laravel-event-sourcing](https://github.com/spatie/laravel-event-sourcing) provides event sourcing in Laravel, but it is directly dependent on Laravel. They use a [EloquentStoredEventRepository](https://github.com/spatie/laravel-event-sourcing/blob/main/src/StoredEvents/Repositories/EloquentStoredEventRepository.php) to handle persistence using a `StoredEvent` Eloquent model.

**Comment by dgafka** [comment]
Laravel's Connection does provide native connection, however due to latest changes to Dbal it required different approach.
This is now fixed starting from `1.31.0`. :)
